### PR TITLE
Feat: add fees only mode filter

### PR DESCRIPTION
**Pull request by RantoniainaAmbi**
Groupe 6:
Description
This PR introduces a fees-only mode filter to the API.

Configuration
The filter's behavior is dynamic and is enabled by defining the configuration directly in your .env file.

Changes
* feat: Added the fees-only mode filter.
* test: Added unit tests to validate the filter's behavior, including edge cases 
  (feesOnly = false, case sensitivity, prefix boundaries).
* docs: Documented the FEES_ONLY resource.
* fix: Corrected prefix matching to avoid false positives — a URI like "/feesnotreal" 
  was previously treated as an allowed subpath of "/fees".
* refactor: Removed manual constructor using System.getenv(); now relies solely on 
  Spring's @Value injection per review feedback.

### CODE REVIEW

**Comment by Andry-Tokiniaina** [comment]
projet 1: groupe 6: MEP

File: src/main/java/school/hei/haapi/endpoint/rest/security/FeesOnlyFilter.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws ServletException, IOException {
+
+    if (!feesOnly) {
+      filterChain.doFilter(request, response);
+      return;
+    }
+
+    String uri = request.getRequestURI();
+    boolean allowed = ALLOWED_PREFIXES.stream().anyMatch(uri::startsWith);
```
Comment (DyferHerioss) [inline src/main/java/school/hei/haapi/endpoint/rest/security/FeesOnlyFilter.java]: use var

**Review by DyferHerioss — CHANGES_REQUESTED** [review]
Create a test for fees_only = false

File: src/main/java/school/hei/haapi/endpoint/rest/security/FeesOnlyFilter.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  private final boolean feesOnly;
+
+  public FeesOnlyFilter() {
+    this("true".equalsIgnoreCase(System.getenv("FEES_ONLY")));
+  }
+
+  public FeesOnlyFilter(boolean feesOnly) {
+    this.feesOnly = feesOnly;
+  }
```
Comment (DyferHerioss) [inline src/main/java/school/hei/haapi/endpoint/rest/security/FeesOnlyFilter.java:22]: Remove this manual constructor and the String comparison. Spring can inject and convert the FEES_ONLY environment variable directly into a boolean using @Value("${FEES_ONLY:false}"), so there's no need to call System.getenv() and compare it with "true" manually.

**Review by DyferHerioss — CHANGES_REQUESTED** [review]

File: src/main/java/school/hei/haapi/endpoint/rest/security/FeesOnlyFilter.java (hunk 1 above)
Comment (RantoniainaAmbi) [reply src/main/java/school/hei/haapi/endpoint/rest/security/FeesOnlyFilter.java]: Done — using var now.

File: src/main/java/school/hei/haapi/endpoint/rest/security/FeesOnlyFilter.java (hunk 2 above)
Comment (RantoniainaAmbi) [reply src/main/java/school/hei/haapi/endpoint/rest/security/FeesOnlyFilter.java:22]: Added 5 test cases covering feesOnly = false (all-pass behavior, empty/root URIs, filterChain invocation).
Removed — now using only the @Value-injected constructor.

**Comment by RantoniainaAmbi** [comment]
All requested changes addressed individually above. Also found and fixed a prefix-matching bug while writing edge-case tests: uri.startsWith(prefix) could match unintended paths like /feesnotreal. Fixed with exact-match-or-subpath check. Ready for re-review.

File: src/main/java/school/hei/haapi/endpoint/rest/security/SecurityConf.java [hunk 3]
```diff
@@ -1126,11 +1120,6 @@ private BearerAuthFilter bearerFilter(RequestMatcher requiresAuthenticationReque
         (httpServletRequest, httpServletResponse, authentication) -> {});
     bearerFilter.setAuthenticationFailureHandler(
         (req, res, e) ->
-            // note(spring-exception)
-            // issues like when a user is not found(i.e. UsernameNotFoundException)
-            // or other exceptions thrown inside authentication provider.
-            // In fact, this handles other authentication exceptions that are
```
Comment (DyferHerioss) [inline src/main/java/school/hei/haapi/endpoint/rest/security/SecurityConf.java:1159]: why remove the existing comments ? Don't touch the comments already existed.

File: src/test/java/school/hei/haapi/unit/FeesOnlyFilterTest.java [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import school.hei.haapi.endpoint.rest.security.FeesOnlyFilter;
+
+@ExtendWith(MockitoExtension.class)
+class FeesOnlyFilterTest {
+
+  @Mock private FilterChain filterChain;
+
+  @Test
```
Comment (DyferHerioss) [inline src/test/java/school/hei/haapi/unit/FeesOnlyFilterTest.java:25]: boy scout pattern, use var

**Review by DyferHerioss — CHANGES_REQUESTED** [review]
